src/cow-react/api/proxy/hooks.ts

```diff
@@ -15,8 +17,28 @@ const SUPPORTED_CHAINS: Partial<Record<Chain, SupportedChainId>> = {
 
 const UNSUPPORTED_CHAIN_ID = null
 
-function hasSupportedChainId(token: any): token is FetchTokensResult {
-  return token?.chainId !== UNSUPPORTED_CHAIN_ID
+// This function will verify if the API response matches our expectations.
+// As we are using an external party, and their responses may change arbitrarily,
+// we should ensure that there is a check so that the application can rely on this abstraction.
```

Comment:
In all fairness explaining this might be a bit of an overkill as well, as this is quite a common pattern for a type guard.

Nevertheless, updated it to use JSDoc 🫡